detectable/src/main/java/com/synopsys/integration/detectable/detectables/bitbake/parse/BitbakeGraphTransformer.java

```diff
@@ -73,4 +89,42 @@ public DependencyGraph transform(final BitbakeGraph bitbakeGraph) {
         return dependencyGraph;
     }
 
+    private Optional<ExternalId> generateExternalId(final String name, final String version, final Map<String, BitbakeRecipe> componentLayerMap, final Map<String, Integer> layerPriorityMap) {
+        final BitbakeRecipe bitbakeRecipe = componentLayerMap.get(name);
+        ExternalId externalId = null;
+
+        if (bitbakeRecipe == null) {
+            if (componentLayerMap.containsKey(name)) {
```

Comment:
Can't you not hit this? If the bitbake recipe was null doesn't that mean component layer map doesn't have the key?